ExampleMod/Content/Items/Weapons/ExampleDualUseWeapon

```diff
@@ -0,0 +1,103 @@
+using ExampleMod.Tiles;
+using Microsoft.Xna.Framework;
+using Terraria;
+using Terraria.ID;
+using Terraria.ModLoader;
+using static Terraria.ModLoader.ModContent;
+
+namespace ExampleMod.Items.Weapons
+{
+	  public class ExampleDualUseWeapon : ModItem
+	  {
+		  public override void SetStaticDefaults() {
+			Tooltip.SetDefault("This weapon does something special with <right>.");
+		}
+		public override void SetDefaults() {
+      item.damage = 50;
+			item.melee = true;
+			item.width = 40;
+			item.height = 40;
+			item.useTime = 20;
+			item.useAnimation = 20;
+			item.useStyle = ItemUseStyleID.SwingThrow;
+			item.knockBack = 6;
+			item.value = 10000;
+			item.rare = ItemRarityID.Green;
+			item.UseSound = SoundID.Item1;
+			item.autoReuse = true;
+			item.shoot = ProjectileID.Bee;
+			item.shootSpeed = 5f;
+		}
+
+		public override void AddRecipes() {
+      CreateRecipe()
+				.AddIngredient<ExampleItem>()
+				.AddTile<ExampleWorkbench>()
+				.Register();
+		}
+
+		public override bool AltFunctionUse(Player player) {
+      // Returned true so you can use the AltFunctionUse
+			return true;
+		}
+
+		public override bool CanUseItem(Player player) {
+    // Check for when the player is using the AltFunctionUse
+			if (player.altFunctionUse == 2) {
+				item.useStyle = ItemUseStyleID.Stabbing;
+				item.useTime = 20;
+				item.useAnimation = 20;
+				item.damage = 50;
+				item.shoot = ProjectileID.Bee;
+			}
+			else {
+      // If not using the AltFunctionUse, use your weapon's default stats, this else block is important because the changes wouldn't revert otherwise
```

Comment:
please try and use terms that describe the function, rather than implementation. `AltFunctionUse` means nothing to someone not familiar with the vanilla variable name. The comments here should explicitly state what `AltFunctionUse` is, (right click) and why the number 2 is meaningful here